app/assets/javascripts/discourse/app/instance-initializers/markdown-tabs.js

```diff
@@ -0,0 +1,50 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+
+function initializeMarkdownTabs(api) {
+  api.decorateCooked(
+    ($elem) => {
+      const tabs = $elem[0].querySelectorAll(".markdown-tabs");
+      if (!tabs.length) {
+        return;
+      }
+
+      tabs.forEach((tabContainer) => {
+        const tabButtons = tabContainer.querySelectorAll(".markdown-tab");
+        const panels = tabContainer.querySelectorAll(".markdown-tab-panel");
+
+        tabButtons.forEach((tab) => {
+          tab.addEventListener("click", (e) => {
```

Comment:
we never remove the listener here

Reply:
I don't think we need to, it will get GCd when the element is removed. If you feel like we need it though I am sure we can figure something out.

Reply:
I don't think that is that simple, because here your closure is holding references to DOM elements which are going to be detached making them way harder to get GCd. At least that has always been my understading.

Is it going to cause massive issues? Probably not.

But this kind of mistakes tend to accumulate, I think it's better to try to make it clean from start when we can see them very clearly, to avoid one day having to decipher memory leaks.

We generally use `api.cleanupStream(handler)` in this situation.